extensions/amp-form/0.1/test/test-amp-form.js

```diff
@@ -424,32 +426,39 @@ describe('amp-form', () => {
       errorTemplate.content.appendChild(
           document.createTextNode('Error: {{message}}'));
       errorContainer.appendChild(errorTemplate);
-      const renderedTemplate = document.createElement('div');
+      let renderedTemplate = document.createElement('div');
       renderedTemplate.innerText = 'Error: hello there';
-      let fetchJsonRejecter;
       sandbox.stub(ampForm.xhr_, 'fetchJson')
-          .returns(new Promise((unusedResolve, reject) => {
-            fetchJsonRejecter = reject;
-          }));
+          .returns(Promise.reject({responseJson: {message: 'hello there'}}));
       sandbox.stub(ampForm.templates_, 'findAndRenderTemplate')
-          .returns(new Promise(resolve => {
-            resolve(renderedTemplate);
-          }));
+          .returns(Promise.resolve(renderedTemplate));
       const event = {
         stopImmediatePropagation: sandbox.spy(),
         target: form,
         preventDefault: sandbox.spy(),
       };
-      ampForm.handleSubmit_(event);
-      fetchJsonRejecter({responseJson: {message: 'hello there'}});
-      return timer.promise(5).then(() => {
-        expect(ampForm.templates_.findAndRenderTemplate.called).to.be.true;
-        expect(ampForm.templates_.findAndRenderTemplate.calledWith(
-            errorContainer, {message: 'hello there'})).to.be.true;
-        // Check that form has a rendered div with class .submit-error-message.
-        const renderedTemplate = form.querySelector('[i-amp-rendered]');
-        expect(renderedTemplate).to.not.be.null;
+      let asyncErrorRejecter;
+      const asyncErrorPromise = new Promise((unusedResolve, reject) => {
+        asyncErrorRejecter = reject;
+      });
+
+      // Catch expected rethrown async error.
+      window.onAsyncErrorForTesting(e => {
```

Comment:
This doesn't seem like the right way to do this. Instead, we could export the module's `asyncErrorsThrownForTesting` and check that after a few seconds.